lib/Deserializer/functions.php

```diff
@@ -346,8 +348,32 @@ function functionCaller(Reader $reader, callable $func, string $namespace)
     }
 
     $funcArgs = [];
+    /**
+     * Even if $func gets "exploded" into an array, that array is still "callable"
+     * It should have elements that are the class and the function/method in the class.
+     * The declaration here helps phpstan to understand.
+     * We check that the class/method/function exists before passing $func to create the Reflection objects.
+     *
+     * @var callable $func
+     */
     $func = is_string($func) && false !== strpos($func, '::') ? explode('::', $func) : $func;
-    $ref = is_array($func) ? new \ReflectionMethod($func[0], $func[1]) : new \ReflectionFunction($func);
+    if (is_array($func)) {
+        if (\class_exists($func[0]) && \method_exists($func[0], $func[1])) {
```

Comment:
after seeing this big block of code, I got the idea that we maybe can just use `is_callable()`?

Reply:
I rewrote the block so it starts with "if is_callable".
Maybe that makes it a bit easier to read?
And it means that there does not need to be a PHPdoc of `$func` declaring that it is `callable`

I still need to check the detail of what sort of callable it is, to choose whether to call `\ReflectionMethod` or `\ReflectionFunction`

Reply:
looks good, thanks